supervisor/http.py

```diff
@@ -663,7 +659,7 @@ def more(self):
             self.file.seek(-bytes_added, 2)
             bytes = self.file.read(bytes_added)
             self.sz = newsz
-            return bytes
+            return as_string(bytes)
```

Comment:
Although I don't mind the other changes, they may or may not be necessary, it is in fact this change that solves the actual bug.